### PR TITLE
Home task 4 for review

### PR DESCRIPTION
**Pull request by NatalliaSem**

### CODE REVIEW

File: HT-2.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import random
```
Comment (Verantina) [inline HT-2.py:1]: What is the reason of having this file here? As I understand, that is just a copy of 2nd homework?

File: HW_4_2_1.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            common_dict.setdefault(key_final, max_value)
+        if key not in duplicated_keys:  # check if the letter is NOT exists in 'duplicated_keys' list
+            for rand_dict in dictList:  # iterate over each dictionary
+                if key in rand_dict:  # check if the specified key is in the specified dictionary
+                    common_dict.setdefault(key, rand_dict[key])  # insert key and its value in the common_dict
+    return common_dict
+
+
+# call the function with 3 parameters
+common_dict_final = create_one_common_dict(distinct_keys, duplicated_keys, dictList)
```
Comment (Verantina) [inline HW_4_2_1.py:64]: Lets have some logic in using functions. There is no need to call each function separately, you can just call all necessary functions from one you use.

File: HW_4_3.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	tHis iz your homeWork, copy these Text to variable. 
+	You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each existING SENtence and add it to the END OF this Paragraph.
+	it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE. 
+	last iz TO calculate nuMber OF Whitespace characteRS in this Text. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+"""
+
+# print(last_word(string))
+# print(text_correct(string))
+print(new_text(string))
+print(calculate_whitespace(string))
```
Comment (Verantina) [inline HW_4_3.py:43]: You just call functions for the same input string. At the same time, there is a chain of tasks and you should use different arguments in your functions to have the task (the whole task to have the final result) done.

File: HT-2.py (hunk 1 above)
Comment (NatalliaSem) [reply HT-2.py:1]: Discussed

**Pull request merged**